include/processor.hh

```diff
@@ -362,25 +604,50 @@ struct Processor {
 		 * from composition of operations - top sort but no dep. on result_idx_
 		 */
 		Operation *op = program_;
+
 		for(auto it=sorted_nodes.rbegin(); it != sorted_nodes.rend(); ++it) {
-			//se._print_node(*it);
+			// se._print_node(*it);
+			// std::cout << "op points at:" << op << std::endl;
 			ScalarNodePtr  node = *it;
 			switch (node->result_storage) {
 			case constant: {
 				double c_val = *node->get_value();
-				double4 * c_ptr = workspace_.vector[node->result_idx_].values;
-				for(uint j=0; j<simd_size; ++j)
-					c_ptr[0][j] = c_val;
+				double * c_ptr = workspace_.vector[node->result_idx_].values;
+				// std::cout << "node->result_idx_ = " << node->result_idx_ << std::endl;
+				// std::cout << "c_ptr = " << c_ptr << std::endl;
+				
+				for(uint j=0; j<simd_size; ++j) {
+					c_ptr[j] = c_val;
+				}
 				break;}
+
+			case constant_bool:
+			{
+				double c_val = *node->get_value();
+				double * c_ptr = workspace_.vector[node->result_idx_].values;
+				Vec<double> v;
+
+				if (c_val == 0.0) {
+					for(uint j=0; j<simd_size; ++j)
+						c_ptr[j] = v.false_value();
+				}
+				else {
+					for(uint j=0; j<simd_size; ++j)
+						c_ptr[j] = v.true_value();
+				}
+				break;
+			}
 			case value:
-				vec_set(node->result_idx_, (double4 *)node->get_value(), workspace_.vec_subset);
+				for (uint i=0; i < simd_size; i++)
```

Comment:
This loop has no effect. Does `vec_set` set whole simd block or just a single value?
Similar in value_copy and expr_result.